### DISCUSSION TITLE
Google Generative AI

### DISCUSSION THREAD

**Discussion by mesencephalon** [mhaowork/amblegpt#22] [Ideas]
Google's API is free, is there a way to adapt it based on Google's AI API?

**Comment by mhaowork** [comment]
I haven't tried Google's API yet. Does it support image input? It shouldn't be hard. If you know how to do it, it should be very quick to swap the model.

**Reply by mesencephalon** [reply]
It supports, please see this link. I tried but your AmbleGPT is way more sophisticated, however google is free

https://github.com/psilantropy/generativeai/blob/main/automation.yml

https://www.reddit.com/r/homeassistant/comments/1an8pgk/google_generative_ai_and_camera_notifications_are/

https://aistudio.google.com/app/apikey

**Reply by mesencephalon** [reply]
As a note, I don't know how to swap the code from OpenAI to Google. Could you provide some guidance? I am not an expert in coding.
